packages/handlersjs-http/lib/services/sync.service.spec.ts

```diff
@@ -0,0 +1,288 @@
+jest.mock('node-fetch');
+
+import { MemoryStore } from '@digita-ai/handlersjs-core';
+import fetch from 'node-fetch';
+import { SyncService } from './sync.service';
+const { Response } = jest.requireActual('node-fetch');
+
+describe('SyncService', () => {
+
+  const headers = new Headers({
+    'Content-Type': 'application/json',
+    Accept: '*/*',
+  });
+
+  const peers = new Set([ 'peer1.com', 'peer2.com' ]);
+
+  type M = {
+    storage: Set<number>; } & {
+    peers: Set<string>;
+    [key: string]: any;
+  };
+
+  let store: MemoryStore<M>;
+  let syncService: SyncService<number, 'storage', 'peers', M>;
+  let fetchMock: jest.MockedFunction<typeof fetch>;
+
+  // saves the latest "modified since" header from mocked requests
+  let latestModifiedSince: string | undefined;
+
+  beforeEach(() => {
+
+    jest.resetAllMocks();
+
+    fetchMock = (fetch as jest.MockedFunction<typeof fetch>);
+    latestModifiedSince = undefined;
+
+    fetchMock.mockImplementation(async (url, options) => {
+
+      latestModifiedSince = options?.headers['If-Modified-Since'];
+
+      return new Response('[]', { status: 200, headers });
+
+    });
+
+    store = new MemoryStore<M>([
+      [ 'storage', new Set<number>() ],
+      [ 'peers', new Set<string>(peers) ],
+      [ 'other', 'something else' ],
+    ]);
+
+    syncService = new SyncService<number, 'storage', 'peers', M>('storage', 'peers', store);
+
+  });
+
+  interface peerResponseMock {
+    peer: string;
+    httpStatus: number;
+    storages?: number[];
+  }
+
+  // mocks the fetch() function with http responses according to the peerResponseMock interface
+  const mockWithStorages = (... responseMocks: peerResponseMock[]) => {
+
+    fetchMock.mockImplementation(async (url, options) => {
+
+      latestModifiedSince = options?.headers['If-Modified-Since'];
+
+      for (const mockInfo of responseMocks) {
+
+        if (mockInfo.peer === url) {
+
+          return new Response(JSON.stringify(mockInfo.storages ? mockInfo.storages : []),
+            { status: mockInfo.httpStatus, headers });
+
+        }
+
+      }
+
+      return new Response('[]', { status: 200, headers });
+
+    });
+
+  };
```

Comment:
The test file reads easier if the beforeEach block comes directly before the tests, so better put interfaces and constants higher up.